xmtp_mls/src/identity.rs

```diff
@@ -132,6 +132,7 @@ pub enum IdentityError {
 #[derive(Debug, Clone)]
 pub struct Identity {
     pub(crate) inbox_id: InboxId,
+    pub(crate) sequence_id: u64,
```

Comment:
If we're storing it in the database as well, maybe we should just use that as the source of truth. What would happen if the DB says you are on `sequence_id` 3 and the struct says you are on `sequence_id` 2

Reply:
It's populated in client creation and not getting set after and reading from db is a bit expensive.

If we want the source of truth in stored in db, would `installation_keys` be stored in db as well?

Reply:
The `installation_keys` are never going to change. But any time you publish an `IdentityUpdate` to the network the `sequence_id` will need to be updated.

We could update the `Identity` object stored on the client every time you publish an `IdentityUpdate` and not worry about the DB if we had it in a mutex.

I just worry about two sources of truth when the data can change over the life of the client.

Reply:
> We could update the Identity object stored on the client every time you publish an IdentityUpdate and not worry about the DB if we had it in a mutex.

Will do

Reply:
One of the remaining TODOs in this work is that we should really call `load_identity_updates` after `apply_signature_request` so that the local DB is always up to date when you make changes. Maybe you can do that as well.